cs/src/comm/epoxy-transport/EpoxyConnection.cs

```diff
@@ -698,49 +689,54 @@ private void DoDisconnected()
             metrics.Emit(ConnectionMetrics);
         }
 
-        private State? DispatchRequest(EpoxyHeaders headers, ArraySegment<byte> payload, ArraySegment<byte> layerData)
+        private State? DispatchRequest(EpoxyHeaders headers, EpoxyProtocol.MessageData messageData, ArraySegment<byte> layerData)
         {
-            if (headers.error_code != (int)ErrorCode.OK)
-            {
-                logger.Site().Error("{0} Received request with a non-zero error code. Conversation ID: {1}",
-                    this, headers.conversation_id);
-                protocolError = ProtocolErrorCode.PROTOCOL_VIOLATED;
-                return State.SendProtocolError;
-            }
-
             Task.Run(async () =>
             {
                 var totalTime = Stopwatch.StartNew();
                 var requestMetrics = Metrics.StartRequestMetrics(ConnectionMetrics);
                 var receiveContext = new EpoxyReceiveContext(this, ConnectionMetrics, requestMetrics);
 
-                IMessage request = Message.FromPayload(Unmarshal.From(payload));
-                IBonded bondedLayerData = (layerData.Array == null) ? null : Unmarshal.From(layerData);
-
-                ILayerStack layerStack;
-                Error layerError = parentTransport.GetLayerStack(requestMetrics.request_id, out layerStack);
-
-                if (layerError == null)
-                {
-                    layerError = LayerStackUtils.ProcessOnReceive(
-                            layerStack, MessageType.Request, receiveContext, bondedLayerData, logger);
-                }
-
+                ILayerStack layerStack = null; // CHWARR TODO: is not giving layers a chance on this response okay?
```

Comment:
Yes I agree. A request received with an error in it shouldn't hit layers on the way in and shouldn't hit layers on the way out -- it is a broken request.